File: C/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int countLetters(string text);
int countWords(string text);
int countSentences(string text);

int main(void)
{

    string text = get_string("Text: ");
    int letters = countLetters(text);
    int words = countWords(text);
    int sentences = countSentences(text);

    printf("%i\n", letters);
    printf("%i\n", words);
    printf("%i\n", sentences);

    float l = (float) letters / (float) words * 100;
    float s = (float) sentences / (float) words * 100;

    int index = round(0.0588 * l - 0.296 * s - 15.8);
    // printf("%i\n", index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int countLetters(string text)
{
    int l = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]) && text[i] != 32)
        {
            l++;
        }
    }
    return l;
}
int countWords(string text)
{
    int w = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isblank(text[i]))
        {
            w++;
        }
    }
    return w + 1;
}
int countSentences(string text)
{
    int s = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            s++;
        }
    }
    return s;
}

File: PYTHON/readability.py
from cs50 import get_string


def countLetters(text):
    l = 0
    for char in text:
        if char.isalpha():
            l += 1
    return l


def countWords(text):
    w = 0
    for char in text:
        if char.isspace():
            w += 1
    return w+1


def countSentences(text):
    s = 0
    for char in text:
        if char in ['!', '.', '?']:
            s += 1
    return s


text = get_string("Text: ")

letters = countLetters(text)
words = countWords(text)
sentences = countSentences(text)

l = letters / words*100
s = sentences / words*100

index = round(0.0588 * l - 0.296 * s - 15.8)

if (index < 1):
    print("Before Grade 1")
elif (index > 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")
